src/kademlia/dht_storage.cpp

```diff
@@ -251,7 +251,7 @@ namespace
 
 					// pick this peer with probability
 					// <peers left to pick> / <peers left in the set>
-					if (random(uint32_t(candidates--)) > to_pick)
+					if (int(random(std::uint32_t(candidates--))) > to_pick)
```

Comment:
hm.. I would have expected making ``to_pick`` unsigned would be more appropriate. ``random()`` will return independent bits as far as I understand, and there's a 50% chance the most significant bit is set. I think this needs to be an unsigned comparison (which it is currently)